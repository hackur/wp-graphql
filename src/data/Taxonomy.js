import { toGlobalId } from 'graphql-relay';

const path = process.env.WP_TAXONOMIES_ENDPOINT || 'graphql/v1/taxonomies';

class Taxonomy {
  getID() {
    return toGlobalId(this.constructor.name, this.slug);
  }

  static getEndpoint() {
    return path;
  }
}

export default Taxonomy;
