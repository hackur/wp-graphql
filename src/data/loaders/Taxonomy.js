import Dataloader from 'dataloader';
import fetchData from 'data/utils';
import Taxonomy from 'data/Taxonomy';

// there is no batch mechanism on this endpoint
const endpoint = Taxonomy.getEndpoint();

// Dataloader expects IDs that can be read by the REST API

export default function getTaxonomyLoaders() {
  const taxonomyLoader = new Dataloader(slugs =>
    fetchData(endpoint).then(({ data: { body } }) => slugs.map(slug => body[slug]))
  );

  return {
    load: async slug => {
      const data = await taxonomyLoader.load(slug);
      return data ? Object.assign(new Taxonomy(), data) : null;
    },
  };
}
