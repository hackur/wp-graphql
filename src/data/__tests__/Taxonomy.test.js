import Taxonomy from 'data/Taxonomy';

describe('Test Taxonomy data access', () => {
  test('Get endpoint', () => {
    expect(Taxonomy.getEndpoint()).toMatchSnapshot();
  });
});
