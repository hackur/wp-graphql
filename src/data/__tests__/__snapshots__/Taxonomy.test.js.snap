// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Taxonomy data access Get endpoint 1`] = `"graphql/v1/taxonomies"`;
